department-of-veterans-affairs/vets-website
# add shell for combobox for revisedAddDisabilities page

**Note**: Delete the description statements, complete each step. **None are optional**, but can be justified as to why they cannot be completed as written. Provide known gaps to testing that may raise the risk of merging to production.

## Summary

- add a new component, ComboBox, for search functionality on the revisedAddDisabilities page
- The new component makes use of the [VADS Text Input](https://design.va.gov/components/form/text-input)
- This is intended to serve as a shell for follow-up tickets: 
  - https://github.com/department-of-veterans-affairs/vagov-claim-classification/issues/456
  - https://github.com/department-of-veterans-affairs/vagov-claim-classification/issues/460
  - https://github.com/department-of-veterans-affairs/vagov-claim-classification/issues/461

## Related issue(s)

- The changes for this PR are catpured in:
  - https://github.com/department-of-veterans-affairs/vagov-claim-classification/issues/457

## Testing done

- manually testing the revised add disabilities page
  -  enable the revised add disabilties page via http://localhost:3000/flipper/features/disability_526_improved_autosuggestions_add_disabilities_page
  - navigate to VA Benefits and Health Care -> Disability -> File a claim online
  - Revised page should render with the `va-text-input`
- automated unit test for verifying component loaded
- `yarn test:unit src/applications/disability-benefits/all-claims/tests/pages/addDisabilities.unit.spec.jsx`
- note new test on L271 of `addDisabilities.unit.spec.jsx`

## Screenshots

![Screenshot 2024-05-08 at 11 09 52 AM](https://github.com/department-of-veterans-affairs/vets-website/assets/17188013/c356247b-12aa-4be7-a78d-f419feae1699)
![Screenshot 2024-05-08 at 11 10 53 AM](https://github.com/department-of-veterans-affairs/vets-website/assets/17188013/238902f3-7ed9-403f-b36a-7aaf4a917b72)


_Note: This field is mandatory for UI changes (non-component work should NOT have screenshots)._

|         | Before | After |
| ------- | ------ | ----- |
| Mobile  |        |       |
| Desktop |        |       |

## What areas of the site does it impact?

*(Describe what parts of the site are impacted **if** code touched other areas)*

## Acceptance criteria

### Quality Assurance & Testing

- n/a
- n/a

### Error Handling


### Authentication


### :warning: Team Sites (only applies to modifications made to the VA.gov header) :warning:


## Requested Feedback

(OPTIONAL) _What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_
